src/xpk/commands/job.py

```diff
@@ -14,12 +14,87 @@
 limitations under the License.
 """
 
+from ..core.commands import run_command_for_value, run_command_with_updates
 from ..utils.console import xpk_exit, xpk_print
-from ..core.core import add_zone_and_project
 from ..core.app_profile import APP_PROFILE_TEMPLATE_DEFAULT_NAME
-from ..core.commands import (
-    run_command_with_updates,
-)
+from ..core.core import add_zone_and_project
+from ruamel.yaml import YAML
+import re
+import sys
+
+
+def job_info(args):
+  """Run commands obtaining information about a job given by name.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    0 if successful and 1 otherwise.
+  """
+  job_name = args.name
+
+  desc_command = f'kubectl-kjob describe slurm {job_name}'
```

Comment:
For readibility of `job_info`, I'd suggest splitting the command executions - describe, list, get - into functions like `run_describe_slurm_command(args)` that return an `int`.

Reply:
Can I create a helper function run_command_for_value_or_exit? Because this is essentially what's happening, and it's supposed to be taken care of by refactor. If not I would leave it as is and wait for exit codes refactor, because then each of these blocks would transform into simple:
```
some_info = run_command_for_value_or_exit("command text for some info")
```
and they would be simple and readable one liners just with variable names.

Reply:
I was thinking of following a similar approach to what we have in `cluster_create`:

https://github.com/AI-Hypercomputer/xpk/blob/3c3b1b54806cdedffa96bbbfb2df00de1e93d2ca/src/xpk/commands/cluster.py#L63

where the command handler - in that case `cluster_create` is in charge of exits based on the returned code by the functions invoked within it - such as `get_gke_server_config`, `set_cluster_command`, etc.-.

However, it's just a suggestion and we may refactor later if needed, so I'm fine if we leave it as is.

Reply:
I will leave it as it is then, since invoking commands is a subject for refactor in the future, and it will simplify it even more.